blog/2025-07-16-updated-pattern-for-helpers-linking-to-devices.md

```diff
@@ -0,0 +1,39 @@
+---
+author: Erik Montnemery
+authorURL: https://github.com/emontnemery
+title: "Updated guidelines for helper integrations linking to other integration's device"
+---
+
+Helper integrations should not add their own config entries to the source entity's device or to a user selected device. Instead, they should just link their entities to the device of the source entity.
+
+Adding the helper config entry to another integration's device is no longer supported and will stop working in Home Assistant Core 2026.8.
+
+### Background
+
+The architecture proposal [home-assistant/architecture#1226](https://github.com/home-assistant/architecture/discussions/1226) makes device connections and identifiers unique per integration domain instead of globally unique. This is not an issue for most integrations, but helper integrations with config flows are an exception.
+
+### Suggested change
+
+#### Do not add the helper config entry to the source device directly
+
+Do not add the config entry to the source device directly, it means this pattern is no longer allowed
+```py
+device_registry.async_update_device(
+    source_device.id,
+    add_config_entry_id=helper_config_entry.entry_id,
+)
+```
+
+#### Do not add the helper config entry to the source device implicitly
+
+Set `self.device_entry` in the helper entity to link the helper entity to another integration's device; don’t set `self._attr_device_info` and don’t override `device_info` to return identifiers and connections for a device from another integration as that will mean the helper config entry is added to the source device.
```

Comment:
```suggestion
Don’t set `self._attr_device_info` and don’t override `device_info` to return identifiers and connections for a device from another integration as that will mean the helper config entry is added to the source device.

Instead, set `self.device_entry` in the helper entity to link the helper entity to another integration's device; 
```

Since the title is "Do not .....", it makes sense to start the section what what not to do now.